database/deployment_test.go

```diff
@@ -0,0 +1,174 @@
+		Commit:      sql.NullString{String: "", Valid: false},
+		Ref:         sql.NullString{String: "", Valid: false},
+		Task:        sql.NullString{String: "", Valid: false},
+		Target:      sql.NullString{String: "", Valid: false},
+		Description: sql.NullString{String: "", Valid: false},
+		Payload:     nil,
+		CreatedAt:   sql.NullInt64{Int64: 0, Valid: false},
+		CreatedBy:   sql.NullString{String: "", Valid: false},
+		Builds:      nil,
+	}
+
+	// setup tests
+	tests := []struct {
+		deployment *Deployment
+		want       *Deployment
+	}{
+		{
+			deployment: testDeployment(),
+			want:       testDeployment(),
+		},
+		{
+			deployment: d,
+			want:       nil,
+		},
+		{
+			deployment: new(Deployment),
+			want:       want,
+		},
+	}
+
+	// run tests
+	for _, test := range tests {
+		got := test.deployment.Nullify()
+
+		if !reflect.DeepEqual(got, test.want) {
+			t.Errorf("Nullify is %v, want %v", got, test.want)
+		}
+	}
+}
+
+func TestDatabase_Deployment_ToLibrary(t *testing.T) {
+	want := new(library.Deployment)
+	want.SetID(1)
+	want.SetNumber(1)
+	want.SetRepoID(1)
+	want.SetURL("https://github.com/github/octocat/deployments/1")
+	want.SetCommit("1234")
+	want.SetRef("refs/heads/main")
+	want.SetTask("deploy:vela")
+	want.SetTarget("production")
+	want.SetDescription("Deployment request from Vela")
+	want.SetPayload(raw.StringSliceMap{"foo": "test1"})
+	want.SetCreatedAt(1)
+	want.SetCreatedBy("octocat")
+	want.SetBuilds(nil)
+
+	got := testDeployment().ToLibrary(nil)
+	if diff := cmp.Diff(got, want); diff != "" {
+		t.Errorf("(ToLibrary: -want +got):\n%s", diff)
+	}
+}
+
+func TestDatabase_Deployment_Validate(t *testing.T) {
+	// setup types
+	tests := []struct {
+		failure    bool
+		deployment *Deployment
+	}{
+		{
+			failure:    false,
+			deployment: testDeployment(),
+		},
+		{ // no number set for deployment
+			failure: true,
+			deployment: &Deployment{
+				ID:     sql.NullInt64{Int64: 1, Valid: true},
+				RepoID: sql.NullInt64{Int64: 1, Valid: true},
+			},
+		},
+		{ // no repoID set for deployment
+			failure: true,
+			deployment: &Deployment{
+				ID:     sql.NullInt64{Int64: 1, Valid: true},
+				Number: sql.NullInt64{Int64: 1, Valid: true},
+			},
+		},
+	}
+
+	// run tests
+	for _, test := range tests {
+		err := test.deployment.Validate()
+
+		if test.failure {
+			if err == nil {
+				t.Errorf("Validate should have returned err")
+			}
+
+			continue
+		}
+
+		if err != nil {
+			t.Errorf("Validate returned err: %v", err)
+		}
+	}
+}
+
+func TestDatabase_DeploymentFromLibrary(t *testing.T) {
+	d := new(library.Deployment)
+	d.SetID(1)
+	d.SetNumber(1)
+	d.SetRepoID(1)
+	d.SetURL("https://github.com/github/octocat/deployments/1")
+	d.SetCommit("1234")
+	d.SetRef("refs/heads/main")
+	d.SetTask("deploy:vela")
+	d.SetTarget("production")
+	d.SetDescription("Deployment request from Vela")
+	d.SetPayload(raw.StringSliceMap{"foo": "test1"})
+	d.SetCreatedAt(1)
+	d.SetCreatedBy("octocat")
+	d.SetBuilds(nil)
```

Comment:
I'd like to see some unit testing on when this field is actually populated. Just a simple build partial or two should work as long as the ID is set.